Sources/BraveWallet/WalletUserAssetManager.swift

```diff
@@ -9,6 +9,13 @@ import BraveCore
 import Preferences
 import CoreData
 
+public protocol WalletUserAssetDataObserver: AnyObject {
+  /// This will be trigger when a balance refresh has been finished
+  func cachedBalanceRefreshed()
+  /// This will be trigger when user asset is updated
+  func userAssetUpdated()
+}
+
 public protocol WalletUserAssetManagerType: AnyObject {
```

Comment:
not a blocker but it'd be nice to get async/await versions of these functions with completion closures for use in Wallet

Reply:
created #8747 for future improvement.